### PR TITLE
Cleanup according to PHPStorm inspection part 4: Fix potential bugs

### PR DESCRIPTION
**Pull request by MrPetovan**
Redux of #6401
Part of #4176
Related to #6404, #6406, #6407

This is the real piece de resistance of this string of PRs. Most of these changes are expected to change Friendica behavior one way or another. Because of this, changes have been neatly broken down into individual commit for your reviewing pleasure/reverting convenience.

This is also the PR that should bring the most shame to Friendica developers given the outlandish nature of some of the mistakes that were made.

### CODE REVIEW

File: mod/directory.php [hunk 1]
```diff
@@ -111,9 +111,12 @@ function directory_content(App $a)
 			$photo = 'photo';
 		}
 
+		$entries = [];
+
 		while ($rr = DBA::fetch($r)) {
 			$itemurl= '';
 
+		while ($rr = DBA::fetch($r)) {
```
Comment (annando) [inline mod/directory.php]: This one confuses me. I don't think that you can easily double fetch the same resource.

File: mod/display.php [hunk 2]
```diff
@@ -80,10 +80,10 @@ function display_init(App $a)
 
 		// Is it an item with uid=0?
 		if (!DBA::isResult($item)) {
-			$item = Item::selectFirstForUser(local_user(), $fields, ['guid' => $a->argv[1], 'private' => [0, 2], 'uid' => 0]);
+			$item = Item::selectFirst($fields, ['guid' => $a->argv[1], 'private' => [0, 2], 'uid' => 0]);
```
Comment (annando) [inline mod/display.php]: Why this change? The "selectFirstForUser" respects possibly hidden public content for the given user.

File: mod/display.php [hunk 3]
```diff
@@ -80,10 +80,10 @@ function display_init(App $a)
...
 		}
 	} elseif (($a->argc == 3) && ($nick == 'feed-item')) {
-		$item = Item::selectFirstForUser(local_user(), $fields, ['id' => $a->argv[2], 'private' => [0, 2], 'uid' => 0]);
+		$item = Item::selectFirst($fields, ['id' => $a->argv[2], 'private' => [0, 2], 'uid' => 0]);
```
Comment (annando) [inline mod/display.php]: Same here.

File: mod/display.php [hunk 4]
```diff
@@ -96,7 +96,7 @@ function display_init(App $a)
 	}
 
 	if ($item["id"] != $item["parent"]) {
-		$item = Item::selectFirstForUser(local_user(), $fields, ['id' => $item["parent"]]);
+		$item = Item::selectFirstForUser($item_user, $fields, ['id' => $item["parent"]]);
```
Comment (annando) [inline mod/display.php]: I think the ```local_user()``` should stay here, since the whole blocking needs to be done for the given viewer.

File: mod/directory.php (hunk 1 above)
Comment (MrPetovan) [reply mod/directory.php]: This looks like an artefact of cherry-picking/rebasing/merging across four different branches, thanks for the keen eye.

File: mod/display.php (hunk 2 above)
Comment (MrPetovan) [reply mod/display.php]: Isn't the `uid` parameter (set in the condition) immediately overwritten in `selectFirstForUser`? This is one of the most contentious changes I wanted to talk to you about.

File: mod/display.php (hunk 4 above)
Comment (MrPetovan) [reply mod/display.php]: Otherwise the `$item_user` isn't used. It is one of the prospective change that makes more sense than removing the variable, but maybe it was just a development artifact.

**Comment by MrPetovan** [comment]
Conflicts fixed.

File: mod/display.php (hunk 2 above)
Comment (annando) [reply mod/display.php]: We always should use ```selectFirstForUser``` or ```selectForUser``` when we are fetching data for display reasons for a specific user. The difference between the calls with and without ```ForUser``` is that we additionally are querying the 'user-item' table where we store things like hiding posts for specific users.

File: mod/display.php (hunk 4 above)
Comment (annando) [reply mod/display.php]: Yeah, possibly we can replace ```local_user()``` with ```$item_user``` here.

File: mod/display.php (hunk 2 above)
Comment (MrPetovan) [reply mod/display.php]: In this case, look at the comment `// Is it an item with uid=0?`, is it compatible with a `selectFirstForUser` call?

Comment (annando) [reply mod/display.php]: Absolutely. The item user has nothing to do with the viewers user id. In fact it is especially meant for posts with uid=0.

Comment (annando) [reply mod/display.php]: I'm waiting with accepting this PR until this line had been changed back.

Comment (MrPetovan) [reply mod/display.php]: Reverted.

File: mod/display.php (hunk 3 above)
Comment (MrPetovan) [reply mod/display.php]: Reverted.
